### ISSUE TITLE
fix: VoiceOver keyboard traps using slots in fast-accordion-item

### ISSUE THREAD

**Issue by MaiLeeTR** [microsoft/fast#6489]
# 🐛 Bug Report

When a user turns on VoiceOver and uses the keyboard to navigate an expanded accordion item with an end slot, they're not able to use the right arrow key to access the panel content. Alternatively, a user is also not able to access the first accordion item panel content if they try to use the left arrow key when they're focused on the second accordion item heading.

## 💻 Repro or Code Sample

1. In Chrome, select [accordion with start and end slots](https://explore.fast.design/components/fast-accordion)
2. Turn on VoiceOver
3. Use the Tab key to navigate to the first accordion item, which will announce "Accordion one, collapsed, button, group"
4. Use the Space key to open the accordion item
5. Use the right arrow key to try to navigate to the panel content 
6. VoiceOver will read the icon "button, group" and further attempts to use the right arrow key to navigate to the panel content is prevented
7. Use the Tab key to move focus onto the second accordion item heading
8. Use the left arrow key to try to navigate to the first accordion item panel content
9. The checkbox is read and further attempts to use the left arrow key to navigate to the previous content is prevented

## 🤔 Expected Behavior

Using VoiceOver and the keyboard, a user should be able to use the right arrow key to access the icon and then use the right arrow key again to access the expanded panel content. They should also be able to use the left arrow key to access the previous expanded content panel.

If the above repro steps are tried with the default accordion example, a user is able to access the panel content with the right arrow key after it's been expanded and also access the previous expanded panel content with the left arrow key.

## 😯 Current Behavior

Any user with VoiceOver turned on and using the keyboard to navigate the accordion component with start and end slots can't access the panel content.

## 🔦 Context

Keyboard traps are a huge issue for those who use the keyboard to navigate, and they're not able to have access to all of the information provided in the panel content. 

We are using the end slot for a badge and noticed this issue upon testing with VoiceOver.

<img width="377" alt="Screen Shot 2022-10-31 at 10 23 25 AM" src="https://user-images.githubusercontent.com/55507048/199044689-409ca14f-9d23-42cf-8e8b-9a47b26bb9bb.png">

## 🌍 Your Environment

* macOS Monterey 12.6.1
* Chrome 106
* VoiceOver

**Comment by EisenbergEffect** [comment]
Thanks for reporting this @MaiLeeTR !
/cc @chrisdholt for triage/investigation.

**Comment by chrisdholt** [comment]
@MaiLeeTR I think I need more insight here on what is expected. Keyboard follows the ARIA Practices guidance detailed here, but AT (screen readers such as Voice Over) do not directly follow that and AT has multiple shortcuts and hotkeys to scan and process content well beyond what is documented in the APG. Unfortunately what we have to work with for communicating with assistive tech is DOM + ARIA; from what I can tell, we've implemented according to the APG which is supported by other AT. If there is an issue here that is specific to voiceover and doesn't repro with other AT I'm not sure how we can address it in isolation just for VoiceOver...it would seem in that case that it's more likely a VoiceOver issue. Does this repro outside VoiceOver and is this perhaps an issue of the various ways that users can navigate via AT?

**Comment by MaiLeeTR** [comment]
@chrisdholt - Thanks for looking through this and providing additional information on the accessibility implementation. We tested with NVDA, and it doesn't reproduce in NVDA. Our QA accessibility testing team wasn't testing with VoiceOver before, but they are going to start. After their testing, I'll better understand if this issue is more specific to VoiceOver than other ATs and will report back.

**Comment by chrisdholt** [comment]
> @chrisdholt - Thanks for looking through this and providing additional information on the accessibility implementation. We tested with NVDA, and it doesn't reproduce in NVDA. Our QA accessibility testing team wasn't testing with VoiceOver before, but they are going to start. After their testing, I'll better understand if this issue is more specific to VoiceOver than other ATs and will report back.

Thanks @MaiLeeTR, let us know what you find out.

**Comment by MaiLeeTR** [comment]
@MendozaA11y - Please update once the team has done VO testing for accordion. Thank you!

**Comment by chrisdholt** [comment]
> @MendozaA11y - Please update once the team has done VO testing for accordion. Thank you!

Yeah, I think what we may find is that while this may be reproducible in certain scenarios for VoiceOver, it likely may not repro in other AT. We can't target fixes for specific assistive tech, and there isn't any code that I see which is actually "trapping" focus on our end, so I'm not sure we'll be able to address this on our end.

**Comment by chrisdholt** [comment]
Pulling a couple labels while we await investigation given the above notes.

**Comment by chrisdholt** [comment]
@MaiLeeTR given this hasn't had any follow-up I'm going to close this. With that said, please do feel free to reopen this if you feel this is still a bug. As noted above, this may be specific to VoiceOver as an AT. We unfortunately only have DOM + ARIA to target assistive tech so we can't specify for one or the other.
